bundles/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/internal/handler/PHCBridgeHandler.java

```diff
@@ -104,215 +110,405 @@ public void initialize() {
             serialPort.addEventListener(this);
             // activate the DATA_AVAILABLE notifier
             serialPort.notifyOnDataAvailable(true);
-            serialPort.setRTS(true);
 
             // get the output stream
             serialOut = serialPort.getOutputStream();
 
-            sendPorBroadcast((byte) 0xFF);
+            sendPorBroadcast();
 
             byte[] b = { 0x01 };
             for (int j = 0; j <= 0x1F; j++) {
                 serialWrite(buildMessage((byte) j, 0, b, false));
-
             }
             updateStatus(ThingStatus.ONLINE);
 
-            new Thread() {
+            // receive messages
+            new Thread("phc-reseive-serial") {
```

Comment:
Do you really need 3 extra Thread's? 

Are they temporary or will they keep running? If they keep running please create your own `ScheduledThreadPoolExecutor` and schedule them inside that, it will allow you to kill them when you are being disposed.

If they only run for a short period of time they should run in the scheduler which is available in the framework's parent class

Reply:
They will keep runnung. 
I think most of the time two would be enough, but from time to time a lot of messages have to be processed in a short time which until now only worked smoothly with the three threads.